### ISSUE TITLE
If the System Probe Agent or Security Agent is disabled, inventory agent may be producing noisy logs.

### ISSUE THREAD

**Issue by tawAsh1** [DataDog/datadog-agent#24177]
Hi Team,

I've noticed a recent update where the Inventory Agent now requests metadata directly from each Datadog Agent, rather than relying on local metadata cache. 
This change, as detailed in PR #22922, means that Inventory Agents periodically fetch metadata from each Agent.

This approach is designed to retrieve metadata irrespective of an Agent's status (enabled/disabled). 
When an Agent is offline, the system fallbacks to using local metadata. 
However, this process outputs specific log entries that may need attention:

```
2024-03-28 10:36:14 UTC | CORE | INFO | (comp/metadata/inventoryagent/inventoryagentimpl/inventoryagent.go:196 in getCorrectConfig) | could not fetch system-probe process configuration: Could not reach system-probe: Get "http://localhost/config": dial unix /opt/datadog-agent/run/sysprobe.sock: connect: no such file or directory
2024-03-28 10:36:14 UTC | CORE | INFO | (comp/metadata/inventoryagent/inventoryagentimpl/inventoryagent.go:196 in getCorrectConfig) | could not fetch security-agent process configuration: Could not reach security-agent: Get "https://localhost:5010/agent/config": dial tcp 127.0.0.1:5010: connect: connection refused
Make sure the system-probe is running before requesting the runtime configuration and contact support if you continue having issues
Make sure the security-agent is running before requesting the runtime configuration and contact support if you continue having issues
```

The logs are output at every inventory check interval, contributing to noise and potential confusion, especially since they suggest operational actions ("Make sure the foo agent is running before requesting the runtime configuration and contact support if you continue having issues") that may not be relevant at the time.

Could we consider suppressing these logs or modifying the messages for clarity?

Thanks!

**Comment by dustmop** [comment]
Hello @tawAsh1,

Thank you for reporting this issue. We agree that this logging is unnecessary and too frequent, and so have [this PR](https://github.com/DataDog/datadog-agent/pull/24407) that removes these messages. After being merged this change will be included in release 7.54 of the Agent, which we expect to be available some time later in Q2.

**Comment by tawAsh1** [comment]
Thank you very much :+1: 

Just a question about Pull Requests, are the following messages also suppressed?

> Make sure the system-probe is running before requesting the runtime configuration, and contact support if you continue having issues.
> Make sure the security-agent is running before requesting the runtime configuration, and contact support if you continue having issues.

This message probably assumes that the target agent is not running when the user intentionally tries to fetch the runtime configuration using the command.
It may be inappropriate(confusable?) as a message when invoked by an automatic fetch from an inventory agent, without the user's direct involvement.

**Comment by dustmop** [comment]
Correct, that message will no longer be output by the automatic config fetcher.

It was being created in the function `FullConfig` by this line https://github.com/DataDog/datadog-agent/blob/c5e4dd6f9bccf3fbdc23324ab19607d892eadd64/pkg/config/settings/http/client.go#L42, then assigned to the variable `err` at this line https://github.com/DataDog/datadog-agent/blob/e4ba2932bc1015b1251e0f108b2f68db50292b5b/comp/metadata/inventoryagent/inventoryagentimpl/inventoryagent.go#L189. The removed log command was outputting that error message, so since it was removed that behavior should end.

**Comment by tawAsh1** [comment]
Thanks for the explanation! 🙏 🙏

**Comment by dustmop** [comment]
You're welcome! Closing this issue now that the PR has merged, please feel free to reopen if you have any other concerns.

**Comment by runlevel5** [comment]
I can confirm I also run into this same warning with v7.53.0

**Comment by mhmdio** [comment]
I also confirm the same - `datadog agent with ECS fargate`

```
May 06, 2024 at 13:43 (UTC+3:00) | 2024-05-06 10:43:35 UTC \| CORE \| INFO \| (comp/metadata/inventoryagent/inventoryagentimpl/inventoryagent.go:198 in getCorrectConfig) \| could not fetch system-probe process configuration: Could not reach system-probe: Get "http://localhost/config": dial unix /opt/datadog-agent/run/sysprobe.sock: connect: no such file or directory | datadog_agent
May 06, 2024 at 13:43 (UTC+3:00) | Make sure the system-probe is running before requesting the runtime configuration and contact support if you continue having issues | datadog_agent
May 06, 2024 at 13:43 (UTC+3:00) | 2024-05-06 10:43:35 UTC \| CORE \| INFO \| (comp/metadata/inventoryagent/inventoryagentimpl/inventoryagent.go:198 in getCorrectConfig) \| could not fetch security-agent process configuration: Could not reach security-agent: Get "https://localhost:5010/agent/config": dial tcp 127.0.0.1:5010: connect: connection refused | datadog_agent
May 06, 2024 at 13:43 (UTC+3:00) | Make sure the security-agent is running before requesting the runtime configuration and contact support if you continue having issues | datadog_agent


```

**Comment by sgnn7** [comment]
Hi @ghouse13 / @runlevel5 / @mhmdio,

Just as a note, as [@dustmop mentioned](https://github.com/DataDog/datadog-agent/issues/24177#issuecomment-2045314347), the fix will be included in Agent v7.54.x line of releases (next upcoming minor release) so it's expected that you still see these messages with the current releases.
